lib/db/mysql/index.js

```diff
@@ -388,19 +398,62 @@ MysqlStore.prototype = {
     });
   },
 
-  getAccessToken: function getAccessToken(tok) {
-    return this._readOne(QUERY_ACCESS_TOKEN_FIND, [buf(tok)]).then(function(t) {
+  /**
+   * Get an access token by token id
+   * @param id Token Id
+   * @returns {*}
+   */
+  getAccessToken: function getAccessToken(id) {
+    return this._readOne(QUERY_ACCESS_TOKEN_FIND, [buf(id)]).then(function(t) {
       if (t) {
         t.scope = t.scope.split(' ');
       }
       return t;
     });
   },
 
+  /**
+   * Remove token by token id
+   * @param id
+   * @returns {*}
+   */
   removeAccessToken: function removeAccessToken(id) {
     return this._write(QUERY_ACCESS_TOKEN_DELETE, [buf(id)]);
   },
 
+  /**
+   * Get all services that have have non-expired tokens
+   * @param {String} uid User ID as hex
+   * @returns {Promise}
+   */
+  getActiveClientTokensByUid: function getActiveClientTokensByUid(uid) {
+    if (! uid) {
+      return P.reject(new Error('Uid is required'));
+    }
+
+    return this._read(QUERY_ACTIVE_CLIENT_TOKENS_BY_UID, [
+      buf(uid)
+    ]);
+  },
+
+  /**
+   * Delete all non-expired tokens for some clientId and uid.
```

Comment:
non-expired part isn't entirely true. This query deletes **all** tokens for this clientId/uid pair.

Not that it shouldn't, just that the comment isn't in sync.